db-service/test/cqn4sql/where-exists.test.js

```diff
@@ -1321,28 +1322,30 @@ describe('cap issue', () => {
     // the described scenario didn't work because in a localized scenario, the localized `foo`
     // association (pointing to `localized.Foo`) was compared to the non-localized version
     // of the association (pointing to `Foo`) and hence, the alias was not properly replaced
-    const cqn = CQL`SELECT from Foo:boos { ID } where exists foo.specialOwners[owner2_userID = $user.id] or exists foo.activeOwners[owner_userID = $user.id]`
-    cqn.SELECT.localized = true
-    let query = cqn4sql(cqn, model)
-    // cleanup
-    delete cqn.SELECT.localized
-    const localized_ = cds.unfold ? '' : 'localized.'
-    expect(query).to.deep.equal(CQL(`
-    SELECT from localized.Boo as boos { boos.ID }
-        WHERE EXISTS (
-          SELECT 1 from localized.Foo as Foo3 where Foo3.ID = boos.foo_ID
-        ) and
-        (
-          EXISTS (
-            SELECT 1 from localized.Foo as foo where foo.ID = boos.foo_ID
-              and EXISTS ( SELECT 1 from ${localized_}SpecialOwner2 as specialOwners where specialOwners.foo_ID = foo.ID and specialOwners.owner2_userID = $user.id )
+    let q = SELECT.localized `from Foo:boos { ID }
+      where exists foo.specialOwners[owner2_userID = $user.id]
+      or exists foo.activeOwners[owner_userID = $user.id]
+    `
+    let q2 = cqn4sql(q, model)
+    expect(cds.clone(q2)).to.deep.equal(CQL(`
+      SELECT from localized.Boo as boos { boos.ID } WHERE EXISTS (
+        SELECT 1 from localized.Foo as Foo3 WHERE Foo3.ID = boos.foo_ID
+      ) AND (
+        EXISTS (
+          SELECT 1 from localized.Foo as foo WHERE foo.ID = boos.foo_ID AND EXISTS (
+            SELECT 1 from ${transitive_?'localized.':''}SpecialOwner2 as specialOwners
+            WHERE specialOwners.foo_ID = foo.ID and specialOwners.owner2_userID = $user.id
           )
-          or EXISTS (
-            SELECT 1 from localized.Foo as foo2 where foo2.ID = boos.foo_ID
-              and EXISTS ( SELECT 1 from ${localized_}ActiveOwner as activeOwners where activeOwners.foo_ID = foo2.ID and activeOwners.owner_userID = $user.id )
+        )
+        OR
+        EXISTS (
+          SELECT 1 from localized.Foo as foo2 WHERE foo2.ID = boos.foo_ID AND EXISTS (
```

Comment:
dont we need the ternary for this `localized` as well?
```suggestion
          SELECT 1 from ${transitive_?'localized.':''}Foo as foo2 WHERE foo2.ID = boos.foo_ID AND EXISTS (
```

Reply:
as above